projects/scion/microfrontend-platform/src/lib/host/client-registry/client.constants.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Specifies the interval (in milliseconds) at which connected clients must send a heartbeat to indicate connectivity to the host.
+ *
+ * By default, if not set, a heartbeat interval of 60s is used.
+ */
+export const CLIENT_HEARTBEAT_INTERVAL = Symbol('CLIENT_HEARTBEAT_INTERVAL');
+
+/**
+ * Specifies the delay (in milliseconds) for unregistering a stale client.
+ *
+ * By default, if not set, stale clients are unregistered after 2 seconds.
+ */
+export const STALE_CLIENT_UNREGISTER_DELAY = Symbol('STALE_CLIENT_UNREGISTER_DELAY');
```

Comment:
When we receive a message from a client without `event source` (`window`) set, that client is typically disconnecting from the host. We do not immediately unregister it so it can shut down gracefully, e.g., send messages or unsubscribe from topics. If we were to immediately unregister the client, those requests would be ignored.

Reply:
OK I understand. Wouldn't it be better then to distinguish between _"missing heartbeat"_ and _"missing event source"_?
- _"missing heartbeat"_ is handled by the `Client`, where stale clients can be removed **immediately**.
- _"missing event source"_ is detected by the `MessageBroker`, which queues the removal for the above mentioned reasons.
I mean, if we don't receive heartbeats from a client after 2 heartbeat periods, then why wait for another two seconds?

Reply:
You are right, delaying the removal after an absent heartbeat makes no sense. Changed the implementation to unregister the client immediately when not receiving a heartbeat anymore.